src/test/java/ticTacToe/player/SmartComputerTest.java

```diff
@@ -0,0 +1,52 @@
+package ticTacToe.player;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import ticTacToe.game.Lines;
+import ticTacToe.grid.Grid;
+import ticTacToe.ui.Ui;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+
+public class SmartComputerTest {
+
+    private static SmartComputer computer;
+
+    @BeforeClass
+    public static void newComputer() {
+        computer = new SmartComputer("X", new MoveGenerator());
+    }
+
```

Comment:
I feel like you don't have enough tests for this class. I guess the test coverage is there but I think it's useful to have tests for different game states like blocking a move or creating a fork or blocking a fork, that type of thing. That can help you prevent bugs and gives you more safety when you refactor the SmartComputer class.

Reply:
(I added more tests although this comment is still shown here)